packages/components/src/date-time/date/index.tsx

```diff
@@ -100,139 +61,255 @@ function DatePickerDay( { day, events = [] }: DatePickerDayProps ) {
 export function DatePicker( {
 	currentDate,
 	onChange,
-	events,
+	events = [],
 	isInvalidDate,
 	onMonthPreviewed,
 	startOfWeek = 0,
 }: DatePickerProps ) {
-	const nodeRef = useRef< HTMLDivElement >( null );
-
-	const onMonthPreviewedHandler = ( newMonthDate: Moment ) => {
-		onMonthPreviewed?.( newMonthDate.toISOString() );
-		keepFocusInside();
-	};
-
-	/*
-	 * Todo: We should remove this function ASAP.
-	 * It is kept because focus is lost when we click on the previous and next month buttons.
-	 * This focus loss closes the date picker popover.
-	 * Ideally we should add an upstream commit on react-dates to fix this issue.
-	 */
-	const keepFocusInside = () => {
-		if ( ! nodeRef.current ) {
-			return;
-		}
+	const date = currentDate ? inputToDate( currentDate ) : new Date();
 
-		const { ownerDocument } = nodeRef.current;
-		const { activeElement } = ownerDocument;
-
-		// If focus was lost.
-		if (
-			! activeElement ||
-			! nodeRef.current.contains( ownerDocument.activeElement )
-		) {
-			// Retrieve the focus region div.
-			const focusRegion = nodeRef.current.querySelector(
-				'.DayPicker_focusRegion'
-			);
-			if ( ! ( focusRegion instanceof HTMLElement ) ) {
-				return;
-			}
-			// Keep the focus on focus region.
-			focusRegion.focus();
-		}
-	};
+	const {
+		calendar,
+		viewing,
+		setSelected,
+		setViewing,
+		isSelected,
+		viewPreviousMonth,
+		viewNextMonth,
+	} = useLilius( {
+		selected: [ startOfDay( date ) ],
+		viewing: startOfDay( date ),
+		weekStartsOn: startOfWeek,
+	} );
 
-	const onChangeMoment = ( newDate: Moment | null ) => {
-		if ( ! newDate ) {
-			return;
-		}
+	const [ focusable, setFocusable ] = useState( startOfDay( date ) );
+	const [ isFocusAllowed, setIsFocusAllowed ] = useState( false );
```

Comment:
It'd be great to add comments above these to explain what they do. I had to read through the code to understand that one is for the roving tab index, and the other is to ensure keyboard events are only handled when focus is in the grid.

Reply:
Yeah ok. I always try to lean more on self-documenting variable names but really struggled with these two weird ones.